### DISCUSSION TITLE
Incremental static regeneration: is there a flag passed to getStaticProps to indicate this?

### DISCUSSION THREAD

**Discussion by joelvarty** [vercel/next.js#21178] [Help]
It would be amazing to be able to tell if a call to getStaticProps is from an incremental regeneration check.

Is there a flag that indicates this?

**Comment by lalit2005** [comment]
https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration

Hope this link will help  🙂

**Reply by joelvarty** [reply]
I don’t see anything in the docs that describes how we would know if a the method is being called via build or via incremental static regeneration

**Reply by lalit2005** [reply]
![image](https://user-images.githubusercontent.com/69138026/104849010-8a059280-590d-11eb-94b7-24b5186ebb84.png)

Here the `revalidate` key shows whether the page uses Incremental Static Regeneration or not.
Hope this answers your question...

**Comment by stigkj** [comment]
I guess the easiest thing to do here is to have some environment variable set when you build the app and check for that in the code.

**Reply by joelvarty** [reply]
Good idea - or a file that’s built that we can check.

**Reply by stigkj** [reply]
@joelvarty what are you going to use this for? Because it is also a possibility to specify that a page should not be rendered build time, i.e. it would then render on the first request and get cached. And you can also choose if this first request should block until the call to `getStaticProps` finishes or render a fallback (i.e. `Loading...`) first before calling `getStaticProps`. For more info: https://nextjs.org/docs/basic-features/data-fetching#fallback-pages

**Reply by joelvarty** [reply]
Exactly my point - I want to be able to know during the getStaticProps call whether this page is being rendered as part of the build or as part of the static regeneration.

**Reply by stigkj** [reply]
If you build on for example Travis you could check for [`TRAVIS=true`](https://docs.travis-ci.com/user/environment-variables/#default-environment-variables). Or as you said, check for a file that's built.

**Comment by GodGardensGuns** [comment]
I have use for this on a website powered by a CMS. My client can select what components appear on what page, as well as what data appears in those components.

Some of that data has to be [cached globally](https://flaviocopes.com/nextjs-cache-data-globally/) for architectural reasons. I want to invalidate that cache whenever a page is rebuilt using ISR.

I suppose I could check [whether the Git environment variables are being populated by Vercel](https://vercel.com/docs/concepts/projects/environment-variables#system-environment-variables), as the project is connected to a Git repo and has automatic builds enabled on each commit. I'll have to test this approach. However, I would prefer a flag in the actual code or something generally more explicit.
